src/operations/course.ts

```diff
@@ -0,0 +1,60 @@
+import { get } from 'http';
+import getDb from '../initDB';
+import { Course, CourseCreate, CourseUpdate,} from '../models';
+
+const db = getDb();
+
+export const createCourse = async (course: CourseCreate): Promise<Course> => {
+  return db.course.create({ data: course });
+};
+
+export const getCourseById = async (
+  id: number,
+  ownerId: string
+): Promise<Course> => {
+  try {
+    const course = await db.course.findFirst({ where: { id, ownerId } });
+    if (!course) {
+      throw new Error('Course not found');
+    }
+    return course;
+  } catch (e) {
+    throw new Error('Course not found');
+  }
+};
+
+export const getCoursesByOwnerId = async (
+  ownerId: string
+): Promise<Course[]> => {
```

Comment:
We should add an optional parameter here, the semesterId. I think the API will use it always but we can make it optional to make our database operations more versatile.